pkg/controllers/cloud/configmap_controller.go

```diff
@@ -30,11 +30,39 @@ import (
 	"antrea.io/nephe/pkg/controllers/config"
 )
 
+type ConfigMapHandler func(controllerConfig *config.ControllerConfig)
+
+// controllerConfigType use to identify the fields in ControllerConfig
+type controllerConfigType int
+
 type ConfigMapReconciler struct {
 	client.Client
-	Log              logr.Logger
-	Scheme           *runtime.Scheme
-	ControllerConfig *config.ControllerConfig
+	Log                  logr.Logger
+	Scheme               *runtime.Scheme
+	ControllerConfig     *config.ControllerConfig
+	configmapHandlerFunc map[controllerConfigType][]ConfigMapHandler
+}
+
+const (
+	cloudSyncIntervalConfig controllerConfigType = iota
+)
+
+var (
+	configmapReconciler *ConfigMapReconciler
+)
+
+// getConfigMapControllerInstance gets the ConfigMap controller instance.
+func getConfigMapControllerInstance() *ConfigMapReconciler {
+	if configmapReconciler == nil {
+		configmapReconciler = &ConfigMapReconciler{}
+		configmapReconciler.configmapHandlerFunc = make(map[controllerConfigType][]ConfigMapHandler)
+	}
+	return configmapReconciler
+}
+
+// setConfigMapControllerInstance sets the ConfigMap controller instance.
+func setConfigMapControllerInstance(reconciler *ConfigMapReconciler) {
```

Comment:
Use init function.. Check https://github.com/antrea-io/nephe/blob/main/pkg/controllers/cloud/controller_sync_status.go#L39

Reply:
We require the ConfigMapReconciler object at the init itself. But the ConfigMapReconciler object creates after the main invoked only.

Reply:
Solved.